arches_lingo/src/arches_lingo/components/scheme/report/SchemeLicense.vue

```diff
@@ -1,22 +1,382 @@
+
+import type {
+    ControlledListResult,
+    ControlledListItem,
+    ControlledListItemResult,
+    DataComponentMode,
+    ResourceInstanceReference,
+    ResourceInstanceResult,
+    SchemeInstance,
+    SchemeRights,
+    SchemeRightStatement,
+} from "@/arches_lingo/types";
+import {
+    selectedLanguageKey,
+    NEW,
+    ERROR,
+    VIEW,
+    EDIT,
+    OPEN_EDITOR,
+    UPDATED,
+} from "@/arches_lingo/constants.ts";
+import ResourceInstanceRelationships from "@/arches_lingo/components/generic/ResourceInstanceRelationships.vue";
+import ReferenceDatatype from "@/arches_lingo/components/generic/ReferenceDatatype.vue";
+import type { Language } from "@/arches_vue_utils/types.ts";
+import NonLocalizedString from "@/arches_lingo/components/generic/NonLocalizedString.vue";
+
+onMounted(async () => {
+    getActorOptions();
+    getControlledLists();
+    getSectionValue();
+});
+
+const emit = defineEmits([OPEN_EDITOR, UPDATED]);
+
+const route = useRoute();
+const router = useRouter();
+const toast = useToast();
 const { $gettext } = useGettext();
+
+const actorRdmOptions = ref<ResourceInstanceReference[]>();
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+const rightTypeOptions = ref<ControlledListItem[]>();
+const languageOptions = ref<ControlledListItem[]>();
+const noteOptions = ref<ControlledListItem[]>();
+const metatypesOptions = ref<ControlledListItem[]>();
+const parentExists = ref(false);
+
+withDefaults(
+    defineProps<{
+        mode?: DataComponentMode;
+        tileId?: string | null;
+        args?: Array<object>;
+    }>(),
+    {
+        mode: VIEW,
+        tileId: null, // editor arg specifying what tile to operate on.
+    },
+);
+
+const schemeRights = ref<SchemeRights>();
+const schemeRightStatement = ref<SchemeRightStatement>();
+
+const referenceNodeConfig = [
+    {
+        nodeAlias: "right_type",
+        listRef: rightTypeOptions,
+        listName: "Right Types",
+    },
+    {
+        nodeAlias: "right_statement_language",
+        listRef: languageOptions,
+        listName: "Languages",
+    },
+    {
+        nodeAlias: "right_statement_type",
+        listRef: noteOptions,
+        listName: "note",
+    },
+    {
+        nodeAlias: "right_statement_type_metatype",
+        listRef: metatypesOptions,
+        listName: "Metatypes",
+    },
+];
+
+async function getActorOptions() {
+    const options_person = await fetchPersonRdmSystemList();
+    const options_group = await fetchGroupRdmSystemList();
+    const options = options_person.concat(options_group);
+
+    actorRdmOptions.value = options.map((option: ResourceInstanceResult) => {
+        const result: ResourceInstanceReference = {
+            display_value: option.descriptors[selectedLanguage.value.code].name,
+            resourceId: option.resourceinstanceid,
+            ontologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
+            inverseOntologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
```

Comment:
is there a way to not hardcode UUIDs on the frontend? If not addressed here, could you file a followup ticket for this?

Reply:
The uuids are removed because the ontology on the current models are actually not assigned (I believe they should be).

I agree that they should not be hard-coded here since the graph can change.
We need more discussion where/how it should get. I feel the related resource object should not be built at all in the front-end but I am not sure. Here is a ticket. https://github.com/archesproject/arches-lingo/issues/191